### PR TITLE
Home Work 16

### PR DESCRIPTION
**Pull request by TanyaB777**

### CODE REVIEW

File: src/main/java/hw_16/Test1.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import static com.codeborne.selenide.Selenide.*;
+
+public class Test1 {
+
+    public static void main(String[] args) {
+
+        final String ERROR_MESSAGE = "Невірний формат номеру посилки. Перевірте вказані символи, а також довжину номеру, яка має бути не більше 13 символів.";
+
+        open("https://rozetka.com.ua");
+        sleep(3000);
```
Comment (AntBelyi) [inline src/main/java/hw_16/Test1.java]: we don't use sleep/ remove in all places please

File: src/main/java/hw_16/Test1.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        SelenideElement trackPackageButton = $(".button.button--medium.button--green");
+
+        trackPackageInput.setValue("20450938174406");
+        trackPackageButton.click();
+        sleep(3000);
+
+        SelenideElement errorMessagePar = $("[class*='validation-message']");
+        String errorMessage = errorMessagePar.getText();
+
+        Assert.isTrue(errorMessage.equals(ERROR_MESSAGE), "The 'Wrong format package tracking number' is not displayed", null);
```
Comment (AntBelyi) [inline src/main/java/hw_16/Test1.java]: Do we need null as a third parameter?

File: src/main/java/hw_16/Test1.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+import static com.codeborne.selenide.Selenide.*;
+
+public class Test1 {
+
+    public static void main(String[] args) {
+
+        final String ERROR_MESSAGE = "Невірний формат номеру посилки. Перевірте вказані символи, а також довжину номеру, яка має бути не більше 13 символів.";
+
+        open("https://rozetka.com.ua");
```
Comment (AntBelyi) [inline src/main/java/hw_16/Test1.java:16]: please implement a window maximize

File: src/main/java/hw_16/Test2.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+import static com.codeborne.selenide.Selenide.*;
+
+public class Test2 {
+    public static void main(String[] args) {
+
+        final String HELP_PAGE_TEXT = "Найактуальніше";
+
+        open("https://rozetka.com.ua");
+        sleep(2000);
```
Comment (AntBelyi) [inline src/main/java/hw_16/Test2.java]: sleep should be removed

File: src/main/java/hw_16/Test3.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+        open("https://rozetka.com.ua");
+
+        SelenideElement cartItem = $(".header-actions .header-cart__button");
+        cartItem.click();
+
+        SelenideElement elementEmptyCart = $(".cart-dummy__heading").shouldBe(Condition.visible);
+        SelenideElement cartImage = $("img.cart-dummy__illustration").shouldBe(Condition.visible);
+
+        Assert.isTrue(elementEmptyCart.getText().contains(EMPTY_CART_TEXT), "The cart is broken");
```
Comment (AntBelyi) [inline src/main/java/hw_16/Test3.java]: "The cart is empty" instead of "The cart is broken"

File: src/main/java/hw_16/Test3.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    public static void main(String[] args) {
+        final String EMPTY_CART_TEXT = "Кошик порожній";
+
+        open("https://rozetka.com.ua");
+
+        SelenideElement cartItem = $(".header-actions .header-cart__button");
+        cartItem.click();
+
+        SelenideElement elementEmptyCart = $(".cart-dummy__heading").shouldBe(Condition.visible);
+        SelenideElement cartImage = $("img.cart-dummy__illustration").shouldBe(Condition.visible);
```
Comment (AntBelyi) [inline src/main/java/hw_16/Test3.java]: Do we need this line?

File: src/main/java/hw_16/Test4.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    public static void main(String[] args) {
+
+        open("https://rozetka.com.ua");
+
+        webdriver().driver().getWebDriver().manage().window().maximize();
+
+        SelenideElement searchInput = $x("//input[contains(@class, 'search-form__input')]");
+        searchInput.setValue("4820227283217").pressEnter();
+
+        SelenideElement elementProduct = $("[class='product-button__buy-group']")
```
Comment (AntBelyi) [inline src/main/java/hw_16/Test4.java]: Variable name is bad

File: src/main/java/hw_16/Test4.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+                .shouldBe(Condition.visible);
+        elementProduct.click();
+
+        SelenideElement elementCartPlus = $("[data-testid='cart-counter-increment-button']")
```
Comment (AntBelyi) [inline src/main/java/hw_16/Test4.java]: Variable name is bad

File: src/main/java/hw_16/Test4.java [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+                .shouldBe(Condition.visible);
+        elementCartPlus.click();
+
+        String valueCount = $("[data-testid='cart-counter-input']")
+                .shouldBe(Condition.visible).getAttribute("value");
+
+        System.out.println("Value: " + valueCount);
+
+        Assert.isTrue(valueCount.contains("2"), "Count of products is not correct");
```
Comment (AntBelyi) [inline src/main/java/hw_16/Test4.java]: Why contains but not equal?

File: src/main/java/hw_16/Test1.java (hunk 1 above)
Comment (TanyaB777) [reply src/main/java/hw_16/Test1.java]: fixed

File: src/main/java/hw_16/Test1.java (hunk 2 above)
Comment (TanyaB777) [reply src/main/java/hw_16/Test1.java]: fixed

File: src/main/java/hw_16/Test2.java (hunk 4 above)
Comment (TanyaB777) [reply src/main/java/hw_16/Test2.java]: fixed

File: src/main/java/hw_16/Test3.java (hunk 5 above)
Comment (TanyaB777) [reply src/main/java/hw_16/Test3.java]: fixed

File: src/main/java/hw_16/Test3.java (hunk 6 above)
Comment (TanyaB777) [reply src/main/java/hw_16/Test3.java]: fixed

File: src/main/java/hw_16/Test4.java (hunk 7 above)
Comment (TanyaB777) [reply src/main/java/hw_16/Test4.java]: fixed

File: src/main/java/hw_16/Test4.java (hunk 8 above)
Comment (TanyaB777) [reply src/main/java/hw_16/Test4.java]: fixed

File: src/main/java/hw_16/Test4.java (hunk 9 above)
Comment (TanyaB777) [reply src/main/java/hw_16/Test4.java]: fixed

**Pull request merged**